# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/rental/tests/string.rs
# Commit: 16350e08d34f
# Full Hash: 16350e08d34ff06c7f46745c6b160c7a7bda9140
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2019-12-06 04:43:57
# Regressor Bug: 1586855
# File Overlap Count: 6
# Description:
#   bug 1586855 - incorporate CRLite filters into cert_storage r=jcj,kjacobs
#   
#   This patch implements CRLite lookups for TLS server certificate revocation
#   information in telemetry-only mode. It adds a new preference
#   "security.pki.crlite_mode" to control the behavior of this feature. Setting
# ==============================================================================

diff -r 64fd40663930 -r 16350e08d34f third_party/rust/rental/tests/string.rs
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/third_party/rust/rental/tests/string.rs	Thu Dec 05 22:41:53 2019 +0000
@@ -0,0 +1,30 @@
+#[macro_use]
+extern crate rental;
+
+
+rental! {
+	pub mod rent_string {
+		#[rental(deref_suffix)]
+		pub struct OwnedStr {
+			buffer: String,
+			slice: &'buffer str,
+			slice_2: &'slice str,
+		}
+	}
+}
+
+
+#[test]
+fn new() {
+	let buf = "Hello, World!".to_string();
+	let _ = rent_string::OwnedStr::new(buf, |slice| slice, |slice, _| slice);
+}
+
+
+#[test]
+fn read() {
+	let buf = "Hello, World!".to_string();
+	let rbuf = rent_string::OwnedStr::new(buf, |slice| slice, |slice, _| slice);
+
+	assert_eq!(&*rbuf, "Hello, World!");
+}